### ISSUE TITLE
Panic: "unexpected item body MaybeDefinitionListTitle"

### ISSUE THREAD

**Issue by maxdeviant** [pulldown-cmark/pulldown-cmark#957]
I'm seeing a panic relating to definitions lists when trying to parse this Markdown:

```md
[abc] check `foobar_raz`\n\n Some preamble `foobar_raz`, not `barfoo_raz`\n :D\n\n This should fix:\n\n &gt; Something is wrong!
```

Here is the backtrace:

```
thread 'main' panicked at /Users/maxdeviant/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pulldown-cmark-0.12.1/src/parse.rs:2156:14:
unexpected item body MaybeDefinitionListTitle
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:72:14
   2: pulldown_cmark::parse::item_to_event
             at /Users/maxdeviant/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pulldown-cmark-0.12.1/src/parse.rs:2156:14
   3: <pulldown_cmark::parse::Parser<F> as core::iter::traits::iterator::Iterator>::next
             at /Users/maxdeviant/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pulldown-cmark-0.12.1/src/parse.rs:2180:29
   4: pulldown_cmark::html::HtmlWriter<I,W>::run
             at /Users/maxdeviant/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pulldown-cmark-0.12.1/src/html.rs:93:33
   5: pulldown_cmark::html::write_html_fmt
             at /Users/maxdeviant/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pulldown-cmark-0.12.1/src/html.rs:622:5
   6: pulldown_cmark::html::push_html
             at /Users/maxdeviant/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pulldown-cmark-0.12.1/src/html.rs:546:5
   7: pulldown_cmark_panic_repro::main
             at ./src/main.rs:10:5
   8: core::ops::function::FnOnce::call_once
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/ops/function.rs:250:5
```

Here is a repository that contains a minimal reproduction: https://github.com/maxdeviant/pulldown-cmark-panic-repro

**Comment by Martin1887** [comment]
Thanks for reporting, please check whether it has been solved in `master` branch after merging #949.

**Comment by maxdeviant** [comment]
> Thanks for reporting, please check whether it has been solved in `master` branch after merging #949.

I tested it out against 64cdb1ea988ba1497ec32ec1af3b16ee0ffec3a6 and the panic is still occurring.
